tests/testthat/test-derive_var_ontrtfl.R

```diff
@@ -437,3 +437,37 @@ test_that("derive_var_ontrtfl Test 15: if trt end date is missing, the obs may s
     )
   )
 })
+
+## Test 16: expected deprecation messaging ----
+test_that("derive_var_ontrtfl Test 16: expected deprecation messaging", { # nolint
+  adcm <- tibble::tribble(
+    ~USUBJID, ~ASTDT, ~TRTSDT, ~TRTEDT, ~AENDT,
+    "P01", ymd("2018-03-15"), ymd("2019-01-01"), NA, ymd("2022-12-01"),
+    "P02", ymd("2020-04-30"), ymd("2019-01-01"), NA, ymd("2022-03-15"),
+    "P03", ymd("2020-04-30"), ymd("2019-01-01"), NA, NA,
+  )
+
+  # all flags should be "Y" because span_period flag is TRUE
+  lifecycle::expect_deprecated(
```

Comment:
@pharmaverse/admiral this seems much more better than shifting between `expecting_warning()` and `expect_error()` with the specific `class  = "lifecycle_xxx_deprecated"`  at each cycle of our process, perhaps we should consider this the way moving forward

Reply:
I like the readability of this expectation. BUT, when a function is fully defunct, you'd need to update to `lifecycle::expect_defunct()`. But I still prefer it because it's easier to understand than a class check

Reply:
This actually brings up a good point, I don't think we've been doing the proper "Defunct"-ing ......